src/PlatformCrypto.c

```diff
@@ -87,16 +87,17 @@ bool PltEncryptMessage(PPLT_CRYPTO_CONTEXT ctx, int algorithm, int flags,
         }
 #endif
         size_t encryptedLength = 0;
+        unsigned char tagTemp[16];
         unsigned char * encryptedData = tag;
+        memcpy(tagTemp, tag, tagLength);
         size_t encryptedCapacity = outLength + tagLength;
-        memcpy(encryptedData + inputDataLength, tag, tagLength);
+        memcpy(encryptedData + inputDataLength, tagTemp, tagLength);
```

Comment:
Maybe I'm missing something, but I'm not sure why this `memcpy()` is even needed at all. For encryption, the tag is an output parameter. `mbedtls_cipher_auth_encrypt_ext()` is going to overwrite whatever we wrote there when it writes the real tag value.

Reply:
You're right! I noticed that that line is completely not needed.